app/helpers/api_helper.rb

```diff
@@ -0,0 +1,38 @@
+module ApiHelper
+  def first_url(params)
+    url_for params.slice(*api_links_default_components)
+  end
+
+  def last_url(params)
+    url_for params.slice(*api_links_default_components).merge(page: params[:total_pages])
+  end
+
+  def next_url(params)
+    params[:page] = Integer(params[:page] || 1)
+
+    if params[:page] < params[:total_pages]
+      url_for params.slice(*api_links_default_components).merge(page: params[:page] + 1)
+    else
+      nil
+    end
+  end
+
+  def prev_url(params)
+    params[:page] = Integer(params[:page] || 1)
+
+    # first page of results does not need the :page param to improve caching
+    return first_url(params) if params[:page] == 2
+
+    if params[:page] > 1
+      url_for params.slice().merge(page: params[:page] - 1)
+    else
+      nil
+    end
+  end
```

Comment:
We should use the pagination details from the results object to generate the urls as these have been sanitised - not that it'd be a security issue but we should try to prevent people from shooting themselves in the foot.